src/compose/volume-manager.ts

```diff
@@ -102,7 +102,15 @@ export async function createFromPath(
 	config: Partial<VolumeConfig>,
 	oldPath: string,
 ): Promise<Volume> {
-	const volume = Volume.fromComposeObject(name, appId, config);
+	const volume = Volume.fromComposeObject(
+		name,
+		appId,
+		// We don't have a uuid available here, but we need one to create a volume
+		// from a compose object. We pass uuid as undefined here so that we will
+		// fallback to id comparison for apps
+		undefined as any,
```

Comment:
Do we want to hack around the transpiler to do this or should we be passing in another value like empty string ? I'm not saying that's the correct thing to do but doing undefined as any seems like we're trying to do something that the transpiler knows is wrong.

Reply:
Should we modify the function params to have a default which is undefined ?

Reply:
That's a good question, I think the function needs to assume that uuid is given, since from now on it should always come as part of the target state. The `createFromPath` function is only called for legacy volumes, which is why this workaround is here. I didn't write this part of the code but I think I'm ok with doing it this way